### ISSUE TITLE
Questions about the keypoint numbers and loss functions on KITTI dataset

### ISSUE THREAD

**Issue by SUZhaoyu** [XuyangBai/D3Feat#53]
Hi, thanks for the great work. I came across a problem when trying to reproduce the performance on KITTI dataset. I tried different combinations of keypoint numbers (from 64 to 1024) and loss functions (both circle_loss and desc_loss). However, none of my test was able to converge given more than 256 keypoints. Can I have any suggestions on this? Cuz I notice in the paper, the model is able to train even with 5000 keypoints.

**Comment by SUZhaoyu** [comment]
UPDATE: with all the setup mentioned above, the best success rate I am able to achieve is 95.13%, with RTE=0.25 and RRE=0.96, which leaves a large margin from the paper. I checkout from the previous [commit](https://github.com/XuyangBai/D3Feat/commit/01c61b964b63915cd686427e5b76c4b517ad0194) and trained with the default parameters (1024 keypoints with `desc_loss`), but the model still refuse to converge.
